scripts/package_tests/ep_testing/tests/api.py

```
@@ -128,211 +129,197 @@ def run(self, install_root: str, verbose: bool, kwargs: dict):
-            if msvc_version == 15:
-                if bitness == 'x64':
-                    command_line.extend(['-G', 'Visual Studio 15 Win64'])
-                elif bitness == 'x32':
-                    command_line.extend(['-G', 'Visual Studio 15'])  # defaults to 32
-            elif msvc_version == 16:
-                if bitness == 'x64':
-                    command_line.extend(['-G', 'Visual Studio 16 2019', '-A', 'x64'])  # default to 64, but be explicit
-                elif bitness == 'x32':
-                    command_line.extend(['-G', 'Visual Studio 16 2019', '-A', 'x86'])
-
-            elif msvc_version == 17:
-                if bitness == 'x64':
-                    command_line.extend(['-G', 'Visual Studio 17 2022', '-A', 'x64'])  # default to 64, but be explicit
-                elif bitness == 'x32':
-                    command_line.extend(['-G', 'Visual Studio 17 2022', '-A', 'x86'])
-            else:
-                raise Exception("Unknown msvc_version passed to make_build_dir_and_build")
+def make_build_dir_and_build(cmake_build_dir: str, verbose: bool, bitness: Bitness, msvc_version: MSVC):
+
+    os.makedirs(cmake_build_dir)
+    my_env = os.environ.copy()
+    if platform.system() == "Darwin":  # my local comp didn't have cmake in path except in interact shells
+        my_env["PATH"] = "/usr/local/bin:" + my_env["PATH"]
+    command_line = ["cmake", ".."]
+    is_windows = platform.system() == "Windows"
+    if is_windows:
+        command_line.extend(["-G", msvc_version.generator_name(), "-A", bitness.value])
 
+    try:
         my_check_call(verbose, command_line, cwd=cmake_build_dir, env=my_env)
-        command_line = ['cmake', '--build', '.']
-        if platform.system() == 'Windows':
-            command_line.extend(['--config', 'Release'])
+    except Exception as e:
+        print("C API Wrapper Configuration Failed!")
+        raise e
```

Couple of things happening here:

* One: I try / catch only the check_call. I don't want to swallow other exception. And I clarify the print to let you know it's the CONFIGURATION that failed here (the build later)
* Two: The cmake Generator options are computed from the enums, less error prone.
* Three: I removed Visual Studio 15. It would have forced me to do more complicated things since it doesn't accept a "-A" argument, and it's too old